### DISCUSSION TITLE
it is work on AMD GPU?

### DISCUSSION THREAD

**Discussion by AdokinBodoke** [lllyasviel/Fooocus#488] [Q&A] [answered]
RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx

some idea?

thanks!

**Comment by harold1110** [comment]
No it does not work on amd gpus but i think they are working on it

**Comment by alli1325** [comment]
hope they released it soon

**Comment by yorai1212** [comment]
Really looking for an AMD support.

**Comment by sarthaq** [comment]
Got something? New here

**Comment by mashb1t** [comment, marked as answer]
AMD GPUs are supported and can be used with DirectML (`--directml` in args).
See
- https://github.com/lllyasviel/Fooocus/blob/main/ldm_patched/modules/args_parser.py#L76
- https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#minimal-requirement

**Reply by mashb1t** [reply]
For Linux ROCm support can be used

**Reply by yorai1212** [reply]
Will generations on AMD cards ever be faster?
I hear of people having 20/30 seconds per generation on Nvidia, while for me it's about 3-5 minutes per images or more.

**Reply by mashb1t** [reply]
In general, AMD VRAM usage has recently been cut in half by an update in the backend code, so i'm positive that with some driver updates by AMD and maybe a few more adjustments image generation might be competitive again.

**Reply by yorai1212** [reply]
Okay, so, in my end, there's only:
1. Updating Fooocus whenever there's an update.
2. Updating AMD drivers whenever there's an update as well.
?

**Reply by mashb1t** [reply]
I mean, you can also buy new hardware, but for now that's the gist, yes.

**Reply by mashb1t** [reply]
@yorai1212 i've just read about https://github.com/vosen/ZLUDA?tab=readme-ov-file#faq, but ROCm should be better as it's officially supported by AMD. Give it a try!

**Reply by yorai1212** [reply]
No idea what that ZLUDA is, I'll just wait for the future then I guess, thank you

**Comment by lgwjames** [comment]
if i use --directml my mac on amd card goes mad

**Comment by hqnicolas** [comment]
Working here:
https://github.com/lllyasviel/Fooocus/discussions/2552
